src/PageSwapper.js

```diff
@@ -0,0 +1,144 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { TransitionGroup } from 'react-transition-group';
+import once from 'once';
+import SwapTransition from './SwapTransition';
+import lockSize from './lock-size';
+import { getRandomNodeKey } from './node-key';
+
+export default class PageSwapper extends Component {
+    static propTypes = {
+        node: PropTypes.element.isRequired,
+        nodeKey: PropTypes.string,
+        animation: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+        children: PropTypes.func.isRequired,
+        onSwapBegin: PropTypes.func,
+        onSwapEnd: PropTypes.func,
+    };
+
+    state = {};
+
+    remainingSwapAcks = 0;
```

Comment:
What's the meaning of SwapAcks? 🤔

Reply:
`acks` is an abbreviation of `acknowledgement`. It's essentially a counter of the remaining transitions that are still inflight. In the beginning of a swap, it's set to `2`. Then, when a view finishes animation, the counter decreases. When it reaches `0`, then the swap has ended.

Reply:
Closing this, but feel free to open it again.